boards/arm/arduino_nano_33_ble/board.c

```diff
@@ -9,16 +9,26 @@
 
 static int board_init(const struct device *dev)
 {
-	const struct gpio_dt_spec pull_up =
-		GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), pull_up_gpios);
-
 	ARG_UNUSED(dev);
 
+	int res;
+	const struct gpio_dt_spec pull_up = GPIO_DT_SPEC_GET(DT_PATH(zephyr_user), pull_up_gpios);
+	const struct gpio_dt_spec user_led = GPIO_DT_SPEC_GET(DT_ALIAS(led4), gpios);
```

Comment:
You may want to make both these structures `static` to "convince" the compiler to put them into the `.rodata` section (instead of locating them on the stack as it is usual for automatic variables, at least gcc will do so) and save a few cycles and a few bytes in flash this way.